Create GSoD 2020 Project Proposal (#1305)
alan-turing-institute/the-turing-way: communications/GSOD-applications/GSoD-2020-Project-Proposal.md

# GSoD 2020 Project Proposal

- Title: Embedding accessibility in The Turing Way open source community guidance 
- Mentors:  Malvika Sharan, Kirstie Whitaker
- Applicant: Pooja Gadige

## Abstract

The Turing Way is an Open Source and community-led book project on research reproducibility and best practices in data science. 
This project involves contributors from all around the world in developing learning resources, guidance and recommendations that form chapters for the books on reproducible research, project design, collaboration, communication and ethics in data science. 
At the time of writing this application, The Turing Way community has around 200 contributors who are data scientists, academics, teachers, students, policy-makers and other stakeholders in research of varying technical backgrounds. 
The Turing Way is hosted and developed online on GitHub (https://github.com/alan-turing-institute/the-turing-way) and the book is hosted as a JupyterBook (https://github.com/jupyter/jupyter-book) on Netlify at https://the-turing-way.netlify.com.

> In 2020, The Turing Way experienced its first major turn over in membership of the core contributor community. The project received a new phase of funding, and pivoted from a focus on computational reproducibility to a much broader scope covering project design, scientific communication, ethical data science, and the skills needed to collaborate in diverse and distributed teams. The Turing Way community members are curating information that is already available in blogs and within the documentation of other open source projects to make it easier for new contributors to any project to effectively join community-led projects around the world. The expansion has greatly improved the potential impact of The Turing Way but has also introduced inconsistencies in the way that the existing chapters have been written. The original authors no longer contribute to the project and it can be confusing for new members to follow updated style guides that existing chapters do not follow themselves.

Author: ```suggestion
In 2020, The Turing Way experienced its first major turn over in membership of the core contributor community. 
The project received a new phase of funding and pivoted from a focus on computational reproducibility to a much broader scope covering project design, scientific communication, ethical data science, and the skills needed to collaborate in diverse and distributed teams. 
The Turing Way community members are curating information that is already available in blogs and within the documentation of other open source projects to make it easier for new contributors to any project to effectively join community-led projects around the world. 
The expansion has greatly improved the potential impact of The Turing Way but has also introduced inconsistencies in the way that the existing chapters have been written. 
The original authors no longer contribute to the project and it can be confusing for new members to follow updated style guides that existing chapters do not follow themselves.
```

 
2. Contributing to future maintenance: build a set of templates for different kinds of contributions in the project, which will ensure that current and future contributors maintain this consistency as the project grows and evolves over time.
 
3. Improving the overall accessibility to information: work on the page tags and filters to make it easier for people to find information in the book. This will be an important work towards improving the user-friendliness of the online resources and give easy access to relevant information searched by the readers.

## Project goals and process in detail

This section explains the project goals in detail, provides examples of tasks that each goal will involve and a description of the process that will be applied to achieve them.

> 1. Achieving retrospective consistency
> 
> I will focus on improving the consistency of content that is already in The Turing Way in order to make the book accessible for readers from diverse backgrounds by:
> 
> i. Ensuring structural consistency across the entire book: By identifying the points that are inconsistent in the book at the moment I will identify a list of structural consistencies to ensure that the chapters across the book follow a uniform pattern, and thus enhance the readability for the users. I will introduce appropriate changes into each chapter to enhance the overall readability of The Turing Way’s book chapters. These changes will include using correct and consistent heading levels to structure a page. Validated headings allow users to read the book using an assistive technology device like a screen reader. I will also ensure that the citations and references are consistent throughout to appropriately cite and provide access to further reading on the Turing Way chapter content.
> 
> ii. Fixing features that are no longer supported: In June 2020 The Turing Way was updated to a new version of JupyterBook that is a complete rewrite of the publishing software. There are many features that are no longer compatible and lead to inconsistencies in the book including broken links and unclear formatting. For example, the previous version of JupyterBook supported HTML tags and a mix of Markdown and HTML formatting, which are no longer supported in the current pure python version. Broken links for cross-referenced chapters or sections make navigating between pages in the book difficult. I will use the new feature of unique labels to define headers and subheaders to make it easier for readers to access information that is available in across The Turing Way chapters.
> 
> 2. Contributing towards the future maintenance 
> 
> I will focus on maintaining consistency of content in The Turing Way in order to make contributing to the project more accessible for new community members.
> 

Author: ```suggestion
**1. Achieving retrospective consistency**

I will focus on improving the consistency of content that is already in The Turing Way in order to make the book accessible for readers from diverse backgrounds by:

i. Ensuring structural consistency across the entire book: By identifying the points that are inconsistent in the book at the moment I will identify a list of structural consistencies to ensure that the chapters across the book follow a uniform pattern, and thus enhance the readability for the users. 
I will introduce appropriate changes to each chapter to enhance the overall readability of The Turing Way’s book chapters. 
These changes will include using correct and consistent heading levels to structure a page. 
Validated headings allow users to read the book using an assistive technology device like a screen reader. 
I will also ensure that the citations and references are consistent throughout to appropriately cite and provide access to further reading on the Turing Way chapter content.

ii. Fixing features that are no longer supported: In June 2020 The Turing Way was updated to a new version of JupyterBook that is a complete rewrite of the publishing software. 
There are many features that are no longer compatible and lead to inconsistencies in the book including broken links and unclear formatting. 
For example, the previous version of JupyterBook supported HTML tags and a mix of Markdown and HTML formatting, which is no longer supported in the current pure python version. 
Broken links for cross-referenced chapters or sections make navigating between pages in the book difficult. 
I will use the new feature of unique labels to define headers and subheaders to make it easier for readers to access information that is available across The Turing Way chapters.

**2. Contributing to future maintenance** 

I will focus on maintaining consistency of content in The Turing Way in order to make contributions to the project more accessible for new community members.

```

> i. Ensuring future consistency: I will develop templates to guide authors as they write new content or modify existing chapters. The current template is outdated and does not reflect the style of communication that The Turing Way community has developed over time. The new templates will make improving the book more accessible by helping new contributors write interoperable content from the start. The material will need less effort to edit and review, and will ensure readers benefit from easy searching, smooth navigation, and better understanding of the technical content. The templates will be accompanied by guidelines for authors who want to develop chapter-specific contents or to incorporate assessments for the readers to self-evaluate concepts covered in the chapter. This effort will bolster asynchronous collaboration among authors when writing content for a chapter ensuring that there is a cohesive narrative when it is written by a diverse group of authors from around the world.
> 
> ii. Capturing the ongoing process and workflow within the community: As a community-developed project, The Turing Way allows anyone to contribute to their project based on their availability. Volunteers carry out initiatives including creating tutorials, documenting case studies, and translating the book into different languages. These efforts have grown organically and are being led by different individuals in an independent manner. Their processes are not well documented making it difficult for new community members to join their work. I will maintain the accessibility and consistency of the project by supporting core contributors to document their workflows and link to them from the main user documentation and style guide.
> 
> 3. Improving an overall accessibility of the online book
> 
> I will focus on improving the accessibility of the material as it grows by adding an advanced filtering option to allow browsing its content by topics, relevance or skill level by:
> 
> i. Enhancing the user-friendliness of the online resource: The goal of The Turing Way is to transform data science practices to be more collaborative and inclusive. This means the information in the book must be accessible to everyone across different skill levels, application domains, and personal characteristics including disabilities such as visual impairment. The guidelines and templates I develop will ensure authors write accessible content from the start. Accessible web content can be read consistently from different digital platforms, can be optimized as per the user’s requirement, and is understood clearly by a broad range of readers. I will improve the accessibility of The Turing Way book by designing and developing a responsive layout for when it is accessed on a smartphone or a tablet. The images in the book will adapt to the screens of mobile devices. I will add keyboard navigation support to the book for people who find it hard to use the mouse trackpad and need to navigate the book using a keyboard. These changes will help make the book accessible to a wider audience.

Author: ```suggestion
i. Ensuring future consistency: I will develop templates to guide authors as they write new content or modify existing chapters. 
The current template is outdated and does not reflect the style of communication that The Turing Way community has developed over time. 
The new templates will make improving the book more accessible by helping new contributors write interoperable content from the start. 
The material will need less effort to edit and review and will ensure readers benefit from easy searching, smooth navigation, and a better understanding of the technical content. 
The templates will be accompanied by guidelines for authors who want to develop chapter-specific content or to incorporate assessments for the readers to self-evaluate concepts covered in the chapter. 
This effort will bolster asynchronous collaboration among authors when writing content for a chapter ensuring that there is a cohesive narrative when it is written by a diverse group of authors from around the world.

ii. Capturing the ongoing process and workflow within the community: As a community-developed project, The Turing Way allows anyone to contribute to their project based on their availability. 
Volunteers carry out initiatives including creating tutorials, documenting case studies, and translating the book into different languages. 
These efforts have grown organically and are being led by different individuals in an independent manner. 
Their processes are not well documented making it difficult for new community members to join their work. 
I will maintain the accessibility and consistency of the project by supporting core contributors to document their workflows and link to them from the main user documentation and style guide.

**3. Improving the overall accessibility of the online book**

I will focus on improving the accessibility of the material as it grows by adding an advanced filtering option to allow browsing its content by topics, relevance, or skill level by:

i. Enhancing the user-friendliness of the online resource: The goal of The Turing Way is to transform data science practices to be more collaborative and inclusive. 
This means the information in the book must be accessible to everyone across different skill levels, application domains, and personal characteristics including disabilities such as visual impairment. 
The guidelines and templates I develop will ensure authors write accessible content from the start.
Accessible web content can be read consistently from different digital platforms, can be optimized as per the user’s requirement, and is understood clearly by a broad range of readers. 
I will improve the accessibility of The Turing Way book by designing and developing a responsive layout for when it is accessed on a smartphone or a tablet. 
The images in the book will adapt to the screens of mobile devices. 
I will add keyboard navigation support to the book for people who find it hard to use the mouse trackpad and need to navigate the book using a keyboard. 
These changes will help make the book accessible to a wider audience.
```

> ii. Providing advanced search capacity to present relevant information: Most users visit The Turing Way with a certain question related to data science in mind. I will enhance the current search feature by developing a standardised vocabulary to support the quick discovery of desired content in the book. This feature will facilitate grouping of related content, both in terms of the material that already exists and new content that will be written in the future. The controlled vocabulary will be developed based on the criteria of how relatively popular a search topic or phrase is and how they relate to the different skill levels. Making sure that the work can be quickly accessed by readers as they look for the answers to the question will make The Turing Way as useful as possible around the open source and data science ecosystems.
> 

Author: ```suggestion
ii. Providing advanced search capacity to present relevant information: Most users visit The Turing Way with a certain question related to data science in mind. 
I will enhance the current search feature by developing a standardised vocabulary to support the quick discovery of desired content in the book. 
This feature will facilitate grouping of related content, both in terms of the material that already exists and new content that will be written in the future. 
The controlled vocabulary will be developed based on the criteria of how relatively popular a search topic or phrase is and how they relate to the different skill levels. 
Making sure that the work can be quickly accessed by readers as they look for the answers to the question will make The Turing Way as useful as possible around the open source and data science ecosystems.

```

To bring it home, I would be working on my dissertation over the summer with a focus on ongoing data challenges in NLP, Search, and Dialogue. Part of my work would involve reproducing results from previous projects and building upon them - this would not be possible if these projects were not reproducible, to begin with. Given this, The Turing Way project appeals to me because of its significance to the field of Data Science.

Furthermore, in the near future, I intend to pursue a PhD with a focus in Data Science. As I aspire to become part of the Data Science research community, I would relish any opportunity that allows me to make a contribution ahead of time.

### Why I am the right person

I believe that I am the right candidate for this project because:
As a budding Data Scientist, I appreciate the need for The Turing Way and foresee the impact it will have on the Data Science community. I will also be able to contribute authentic content to The Turing Way based on my experiences.

> In my previous role as a Research Analyst at Guaranty Trust Bank in Nigeria, I gained significant experience in creating and maintaining documentation for projects my team was responsible for. I created customer-facing user-manuals for an app called Orange Books and another called Zumi the SKS Robot. Though these manuals were distributed in print, the final versions can be accessed here and here.

Author: ```suggestion
In my previous role as a Research Analyst at Guaranty Trust Bank in Nigeria, I gained significant experience in creating and maintaining documentation for projects my team was responsible for. I created customer-facing user-manuals for an app called Orange Books and another called Zumi the SKS Robot. Though these manuals were distributed in print, the final versions can be accessed online.
```
